cmd/commands/function.go

```diff
@@ -44,66 +41,27 @@ func Function() *cobra.Command {
-	}
-	sort.Strings(invokerNames)
-
 	command := &cobra.Command{
 		Use:   "create",
 		Short: "Create a new function resource",
-		Long: "Create a new function resource from the content of the provided Git repo/revision or local source.\n" +
-			"\nThe INVOKER arg defines the language runtime and function invoker that is added to the function code in the build step. The resulting image is then used to create a Knative Service (`service.serving.knative.dev`) instance of the name specified for the function. The following invokers are available:\n\n" +
-			strings.Join(invokerNames, "") +
-			"- 'custom': use a custom invoker. Specify with --invoker-url flag\n" +
-			"\nBuildpack based builds support building from local source or within the cluster. Images will be pushed to the registry specified in the image name, unless prefixed with 'dev.local/' in which case the image will only be available within the local Docker daemon.\n" +
-			"\nFrom then on you can use the sub-commands for the `service` command to interact with the service created for the function.\n\n" +
+		Long: "Create a new function resource from the content of the provided Git repo/revision or local source.\n\n" +
+			"The --invoker flag can be used to force the language runtime and function invoker that is added to the function code in the build step. The resulting image is then used to create a Knative Service (`service.serving.knative.dev`) instance of the name specified for the function.\n\n" +
+			"Images will be pushed to the registry specified in the image name, unless prefixed with 'dev.local/' in which case the image will only be available within the local Docker daemon.\n\n" +
+			"From then on you can use the sub-commands for the `service` command to interact with the service created for the function.\n\n" +
 			envFromLongDesc + "\n",
-		Example: `  ` + env.Cli.Name + ` function create node square --git-repo https://github.com/acme/square --image acme/square --namespace joseph-ns
-  ` + env.Cli.Name + ` function create java tweets-logger --git-repo https://github.com/acme/tweets --image acme/tweets-logger:1.0.0`,
+		Example: `  ` + env.Cli.Name + ` function create square --git-repo https://github.com/acme/square --artifact square.js --image acme/square --invoker node --namespace joseph-ns
+  ` + env.Cli.Name + ` function create tweets-logger --git-repo https://github.com/acme/tweets --image acme/tweets-logger:1.0.0`,
+		PreRunE: FlagsValidatorAsCobraRunE(AtLeastOneOf("git-repo", "local-path")),
 		Args: ArgValidationConjunction(
 			cobra.ExactArgs(functionCreateNumberOfArgs),
-			AtPosition(functionCreateInvokerIndex, ValidName()),
 			AtPosition(functionCreateFunctionNameIndex, ValidName()),
 		),
-		PreRunE: func(cmd *cobra.Command, args []string) error {
-			err := flagsValidator(cmd)
-			if err != nil {
-				return err
-			}
-
-			invoker := args[functionCreateInvokerIndex]
-			if invoker != "custom" && createFunctionOptions.InvokerURL != "" {
```

Comment:
What should users who are using customer invokers do instead? Do we need some docs or at least a notice in the release notes saying the function has been dropped?

Reply:
They'll eventually use custom *builders*. Supported for the local case only (via this PR) for now